charts/k8ssandra/templates/_helpers.tpl

```diff
@@ -153,4 +153,39 @@ Set default num_tokens based on the server version
       num_tokens: 16
     {{- end }}
     {{- end }}
-{{- end }}
\ No newline at end of file
+{{- end }}
+
+{{- define "medusa.configMapName" -}}
+{{ .Release.Name }}-medusa
+{{- end }}
+
+{{/*
+Creates Cassandra auth environment variables if authentication is enabled.
+*/}}
+{{- define "medusa.cassandraAuthEnvVars" -}}
+{{- if .Values.cassandra.auth.enabled }}
+  {{- if .Values.medusa.cassandraUser.secret }}
+    {{- nindent 10 "- name: CQL_USERNAME" }}
+    {{- nindent 12 "valueFrom:" }}
+    {{- nindent 14 "secretKeyRef:" }}
+    {{- nindent 16 (print "name: " .Values.medusa.cassandraUser.secret) }}
+    {{- nindent 16 "key: username" }}
+    {{- nindent 10 "- name: CQL_PASSWORD" }}
+    {{- nindent 12 "valueFrom:" }}
+    {{- nindent 14 "secretKeyRef:" }}
+    {{- nindent 16 (print "name: " .Values.medusa.cassandraUser.secret) }}
+    {{- nindent 16 "key: password" }}
+  {{- else }}
+    {{- nindent 10 "- name: CQL_USERNAME" -}}
+    {{- nindent 12 "valueFrom:" }}
+    {{- nindent 14 "secretKeyRef:" }}
+    {{- nindent 16 (print "name: " (include "k8ssandra.clusterName" . ) "-medusa") }}
+    {{- nindent 16 "key: username" }}
+    {{- nindent 10 "- name: CQL_PASSWORD" }}
+    {{- nindent 12 "valueFrom:" }}
+    {{- nindent 14 "secretKeyRef:" }}
+    {{- nindent 16 (print "name: " (include "k8ssandra.clusterName" . ) "-medusa") }}
+    {{- nindent 16 "key: password" }}
+  {{- end -}}
```

Comment:
wouldn't this be nicer as a template?

Reply:
Not sure I follow. Can you explain?

Reply:
I guess what I'm suggesting is that you use actual spaces instead of `nindent ##`.